### PR TITLE
docs: improve custom component dirs example

### PR DESCRIPTION
**Pull request by jd-solanki**
### 🔗 Linked issue

#22690

### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Improves custom dirs example by adding more example and what will be output

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

-----
<a href="https://stackblitz.com/~/github/jd-solanki/nuxt-docs/tree/main">Review PR in StackBlitz Codeflow</a> _Submitted with [StackBlitz Codeflow](https://stackblitz.com/~/github/jd-solanki/nuxt-docs/tree/main)._

### CODE REVIEW

File: docs/2.guide/2.directory-structure/1.components.md [hunk 1]
```diff
@@ -34,8 +34,16 @@ By default, only the `~/components` directory is scanned. If you want to add oth
 ```ts [nuxt.config.ts]
 export default defineNuxtConfig({
   components: [
-    { path: '~/components/special-components', prefix: 'Special' },
+    // ~/components/Btn.vue => <Btn />
+    // ~/components/base/Btn.vue => <BaseBtn />
     '~/components'
```
Comment (danielroe) [inline docs/2.guide/2.directory-structure/1.components.md:39]: this should always appear last

File: docs/2.guide/2.directory-structure/1.components.md [hunk 2]
```diff
@@ -34,8 +34,16 @@ By default, only the `~/components` directory is scanned. If you want to add oth
...
+
+    // ~/components/base/Btn.vue => <Btn />
+    { path: '~/components/base/', pathPrefix: false },
```
Comment (danielroe) [inline docs/2.guide/2.directory-structure/1.components.md:42]: this doesn't really make any difference in the example you use as there is no path refix. Maybe `~/components/base/further/Btn.vue` -> `<Btn />` would be a better example?

Comment (danielroe) [inline docs/2.guide/2.directory-structure/1.components.md:42]: this doesn't really make any difference in the example you use as there is no path refix. Maybe `~/components/base/further/Btn.vue` -> `<Btn />` would be a better example?

File: docs/2.guide/2.directory-structure/1.components.md (hunk 1 above)
Comment (jd-solanki) [reply docs/2.guide/2.directory-structure/1.components.md]: Moved components dir to last 👍🏻

File: docs/2.guide/2.directory-structure/1.components.md (hunk 2 above)
Comment (jd-solanki) [reply docs/2.guide/2.directory-structure/1.components.md:38]: According to me component `~/components/base/Btn.vue` will become `<BaseBtn />`, right?

Comment (danielroe) [reply docs/2.guide/2.directory-structure/1.components.md:38]: Not if you add `~/components/base` - everything in this `path` will be ignored for the purposes of the component name

File: docs/2.guide/2.directory-structure/1.components.md [hunk 3]
```diff
@@ -34,7 +34,17 @@ By default, only the `~/components` directory is scanned. If you want to add oth
 ```ts [nuxt.config.ts]
 export default defineNuxtConfig({
   components: [
+    // ~/calendar-module/components/event/Update.vue => <EventUpdate />
+    { path: '~/calendar-module/components/' },
```
Comment (danielroe) [inline docs/2.guide/2.directory-structure/1.components.md:38]:
```suggestion
    { path: '~/calendar-module/components' },
```

File: docs/2.guide/2.directory-structure/1.components.md [hunk 4]
```diff
@@ -34,7 +34,17 @@ By default, only the `~/components` directory is scanned. If you want to add oth
...
+
+    // ~/user-module/components/account/UserDeleteDialog.vue => <UserDeleteDialog />
+    { path: '~/user-module/components/', pathPrefix: false },
```
Comment (danielroe) [inline docs/2.guide/2.directory-structure/1.components.md:41]:
```suggestion
    { path: '~/user-module/components', pathPrefix: false },
```

File: docs/2.guide/2.directory-structure/1.components.md [hunk 5]
```diff
@@ -34,7 +34,17 @@ By default, only the `~/components` directory is scanned. If you want to add oth
...
+
+    // ~/components/special-components/Btn.vue => <SpecialBtn />
     { path: '~/components/special-components', prefix: 'Special' },
+
+    // ~/components/Btn.vue => <Btn />
```
Comment (danielroe) [inline docs/2.guide/2.directory-structure/1.components.md:46]:
```suggestion
    // It's important that this comes last if you have overrides you wish to apply
    // to sub-directories of `~/components`.
    //
    // ~/components/Btn.vue => <Btn />
```

**Review by danielroe — APPROVED** [review]

**Comment by danielroe** [comment]
Thank you! ❤️

**Pull request merged**